### ISSUE TITLE
Build nightly Windows snapshots?

### ISSUE THREAD

**Issue by emmatyping** [tstellar/actions#1]
Currently https://llvm.org/builds/ provides "periodic" snapshot builds of clang for Windows. It would be awesome if this could be adapted to releasing nightly (or even just weekly!) builds. I tried to start working on that here: https://github.com/ethanhs/build-clang, but that didn't work out. The packaging is done by this script: https://github.com/llvm/llvm-project/blob/master/llvm/utils/release/build_llvm_package.bat

Perhaps an equivalent could be built on top of your work here. Also it could be scheduled to run every <time period> via [`on.schedule`](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#onschedule)

**Comment by tstellar** [comment]
I agree this would be useful.  It looks like the build_llvm_package.bat needs to be updated to pull from GitHub first.

If you want to try to build your own workflow, I would recommend using the setup-windows and install-ninja actions defined in this repository.  We are using these in our CI tests in the release/9.x branch for setting up the windows environment.

**Comment by tstellar** [comment]
Actually these actions are in the official llvm/actions repo now, so it would be better to use those.

**Comment by Trass3r** [comment]
@tstellar btw is there a good reason to use actions for everything?

Especially https://github.com/tstellar/actions/blob/master/build-test-llvm-project/main.js +
```yaml
    - name: Test LLVM
      uses: llvm/actions/build-test-llvm-project@master
      with:
        cmake_args: -G Ninja -DLLVM_ENABLE_PROJECTS="clang;lld;" -DCMAKE_BUILD_TYPE=Release
```
looks like total overkill/boilerplate while a simple run step could be enough and isn't any more hardcoded:
```yaml
    - shell: bash
      run: |
        mkdir build && cd build
        cmake -G Ninja [...] ..
        ninja
```
They support bash on Windows and even python scripts.

**Comment by tstellar** [comment]
@Trass3r I debated a lot whether to use actions or not for various steps.  I agree in some cases it seems like overkill.

For the build-test-llvm-project step, it calls out to https://github.com/tstellar/actions/blob/master/configure-llvm-project/main.js
which adds some operating system specific CMake flags to ensure that ninja and ninja check-all will pass in the GitHub environment.  So it's not just a simple wrapper around cmake.

**Comment by Trass3r** [comment]
I see, thanks. I guess to do that "inline" one could set some env vars depending on the OS.
Did you find any other actions useful for C++ development? There's not much yet. I created a simple [one](https://github.com/Trass3r/setup-cpp) to make it collect compiler messages at least.
